strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/storage/repository/Repository.java

```diff
@@ -328,14 +329,14 @@ public String toString()
         return id;
     }
 
-    public File getTrashDir()
+    public Path getTrashDir()
     {
-        return new File(getBasedir(), ".trash");
+        return Paths.get(getBasedir(), ".trash");
```

Comment:
@carlspring @ptirador 
We shouldn't resolve paths with `Paths.get` for now. There is appropriate method `RepositoryProvider.resolvePath` so we should always use it instead of `Paths.get`.
Also I suggest to remove this methods directly from Repository, will be much better to have this methods within some service, it can be `RepositoryProvider` for example.

Reply:
Yeah, his is what I actually meant. Apologies, Pablo!

Reply:
I have changed the description to be more clear. Once again -- apologies for not having been clear from the start.

@ptirador , would you mind correcting this please?

Reply:
If I move the methods to `RepositoryProvider` service, which has 3 implementations, this wouldn't be a minor refactoring...
For example, in `LayoutProviderRegistry.deleteTrash()`:

`final File basedirTrash = repository.getTrashDir();`

How am I supposed to obtain the `RepositoryProvider` service? Or I'd need to create these methods (getTrashDir, getTempDir) in `LayoutProvider` too?

Reply:
@sbespalov ,

Any advice how to proceed with this?

Reply:
I just thought that there is another way to obtain `trash` and `temp` folders which can suit better in many cases. You can use `RepositoryFileSystem.getTrashPath()` or `RepositoryFileSystem.getTempPath()` for this. So if you have `RepositoryPath` instance then below is the way to get wnat you need:
```
((RepositoryPath)path).getFileSystem().getTempPath();
```